frontend/server/src/DAO/Submissions.php

```diff
@@ -313,4 +313,34 @@ public static function getLatestSubmissions(
             'submissions' => $submissions,
         ];
     }
+
+    /** @return list<array{time: \OmegaUp\Timestamp}> */
+    public static function getLastSubmissionByProblem(
+        int $problemId,
+        ?int $identityId
+    ): array {
+        $params = [$problemId];
+        $sql = '
+            SELECT
+                s.time
+            FROM
+                Submissions s
+            WHERE
+                s.problem_id = ?
+        ';
+
+        if (!is_null($identityId)) {
+            $params[] = $identityId;
+            $sql .= ' AND s.identity_id = ?';
+        }
+
+        $sql .= '
+            ORDER BY
+                s.time DESC
+            LIMIT 1
+        ';
+
+        /** @var list<array{time: \OmegaUp\Timestamp}> */
+        return \OmegaUp\MySQLConnection::getInstance()->GetAll($sql, $params);
```

Comment:
Existe una función que obtiene algún dato único: 
```suggestion
        /** @var \OmegaUp\Timestamp */
        return \OmegaUp\MySQLConnection::getInstance()->GetOne($sql, $params);
```
Así, desde el controlador ya no tienes que acceder al primer elemento del array, sino al valor que te regresa esta función (También habría que arreglar el tipo en la declaración del return).

Reply:
Genial.